### ISSUE TITLE
Inconsistency in `fractions.Fraction()` initialization error with invalid string

### ISSUE THREAD

**Issue by thatbirdguythatuknownot** [python/cpython#114014]
### Bug description:

The `fractions.Fraction()` error message differs when, for example, `123.dd` is done instead of `123.aa`; they should be equal.
```pycon
>>> from fractions import Fraction
>>> Fraction("123.dd")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
    Fraction("123.dd")
  File "C:\Program Files\Python313\Lib\fractions.py", line 251, in __new__
    numerator = numerator * scale + int(decimal)
                                    ^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'dd'
>>> Fraction("123.aa")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
    Fraction("123.aa")
  File "C:\Program Files\Python313\Lib\fractions.py", line 239, in __new__
    raise ValueError('Invalid literal for Fraction: %r' %
ValueError: Invalid literal for Fraction: '123.aa'
```
I discovered this bug while inspecting the rational parsing regex.
https://github.com/python/cpython/blob/dac1da21218a406652b35919aa2118cc32d4c65a/Lib/fractions.py#L57-L69
I think the bug stems from line 65's matching of `d*` instead of `\d*`.

### CPython versions tested on:

3.13

### Operating systems tested on:

Windows

### Linked PRs
* gh-114015
* gh-114023
* gh-114025

**Comment by terryjreedy** [comment]
@mdickinson fractions issue

**Comment by mdickinson** [comment]
Nice find! Agreed that this is a clear bug. Looks like it was introduced in #24622, so the bug is present in 3.11 onwards; the fix should be backported to 3.12 and 3.11.
